consumer.go

```diff
@@ -269,6 +274,30 @@ func (c *Consumer) getQueueLength() (n int64, err error) {
 	return
 }
 
+func (c *Consumer) getNextFilename(meta Meta) (filename string, err error) {
+	var ok bool
+	if filename, ok = c.f.Shift(); ok {
+		return
+	}
+
+	lastFile := makeFilename(c.opts.FullName(), meta.LastProcessedTimestamp, meta.LastProcessedType)
```

Comment:
not sure... do we got it from list to download? 
or from last imported to db?

Reply:
Great question! It's from last processed which we pull from the Meta. So meta controls the state of kiroku and `getNextFilename` is called while under a mutex lock for Meta.

Reply:
I added a comment explaining, let me know if this was helpful